astropy/modeling/physical_models.py

```diff
@@ -0,0 +1,190 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Models that have physical origins.
+"""
+
+import warnings
+from collections import OrderedDict
+
+import numpy as np
+
+from .core import Fittable1DModel
+from .parameters import Parameter
+from astropy import constants as const
+from astropy import units as u
+from astropy.utils.exceptions import AstropyUserWarning
+
+__all__ = ["BlackBody"]
+
+
+class BlackBody(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : :class:`~numpy.ndarray` or :class:`~astropy.units.Quantity`
+        Blackbody temperature.
+
+    scale : :class:`~numpy.ndarray` or :class:`~astropy.units.Quantity`
+        Scale factor
+
+    Notes
+    -----
+
+    Model formula:
+
+        .. math:: B_{\\nu}(T) = A \\frac{2 h \\nu^{3} / c^{2}}{exp(h \\nu / k T) - 1}
+
+    Examples
+    --------
+    >>> from astropy.modeling import models
+    >>> from astropy import units as u
+    >>> bb = models.BlackBody(temperature=5000*u.K)
+    >>> bb(6000 * u.AA)  # doctest: +FLOAT_CMP
+    <Quantity 1.53254685e-05 erg / (cm2 Hz s sr)>
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        from astropy.modeling.models import BlackBody
+        from astropy import units as u
+        from astropy.visualization import quantity_support
+
+        bb = BlackBody(temperature=5778*u.K)
+        wav = np.arange(1000, 110000) * u.AA
+        flux = bb(wav)
+
+        with quantity_support():
+            plt.figure()
+            plt.semilogx(wav, flux)
```

Comment:
Now that you added method for lambda max, maybe plot it as well? :thinking:

Reply:
Done.

Reply:
I'm not sure how useful it is to have plots in the API  docs. How does user normally interact with this? From the terminal/notebook they won't be able to see the plots, do they?